### PR TITLE
Run blocks with boxes checked next to them

### PR DESCRIPTION
**Pull request by fsih**
When a checkbox is checked in the flyout, calculate the value of that block every frame in the VM. Does not yet deal with single sprite scoped variables.

### CODE REVIEW

**Comment by fsih** [comment]
Does this need tests?

File: src/virtual-machine.js [hunk 1]
```diff
@@ -345,6 +346,16 @@ class VirtualMachine extends EventEmitter {
     }
 
     /**
+     * Handle a Blockly event for the flyout to be passed to the monitor container.
+     * @param {!Blockly.Event} e Any Blockly event.
+     */
+    monitorBlockListener (e) {
+        if (['create', 'change'].indexOf(e.type) !== -1) {
```
Comment (rschamp) [inline src/virtual-machine.js]: We should explain this whitelist in the docstring

**Review by rschamp — APPROVED** [review]

**Comment by rschamp** [comment]
It would be great to have tests for this. But since this is still incomplete, we can write them as we finish the feature... we might be making more work for ourselves writing them before we finish figuring out exactly how it should work.

**Comment by mrjacobbloom** [comment]
something to watch out for: right now no reporters should have any side effects, but if custom reporters are implemented that might become an issue

**Comment by rschamp** [comment]
@mrjacobbloom That's a good point! For that reason we will probably not allow monitoring of custom reporters if they are added. (And probably the reason you can't monitor custom reporters in Snap!)

File: src/engine/execute.js [hunk 2]
```diff
@@ -85,8 +87,22 @@ const execute = function (sequencer, thread) {
...
-                runtime.visualReport(currentBlockId, resolvedValue);
+                if (thread.showVisualReport) {
+                    runtime.visualReport(currentBlockId, resolvedValue);
+                }
+
+                if (thread.updateMonitor) {
+                    runtime.updateMonitors([{
+                        id: currentBlockId, // @todo(dd) this will collide if multiple sprites use same block
+                        category: 'data',
+                        label: blockContainer.getOpcode(blockContainer.getBlock(currentBlockId)), // @todo(dd) how to handle translation here?
```
Comment (CosmicWebServices) [inline src/engine/execute.js]: 100:1   error    Line 100 exceeds the maximum line length of 120       max-len

Travis error

**Review by CosmicWebServices — CHANGES_REQUESTED** [review]
Travis does not pass

File: src/virtual-machine.js (hunk 1 above)
Comment (fsih) [reply src/virtual-machine.js]: Done

**Comment by fsih** [comment]
This code won't do anything until the blockly change to fire checkbox events is in. https://github.com/paulkaplan/scratch-blocks/tree/emit-checkbox-event

**Pull request merged**